### DISCUSSION TITLE
How can I have GrapesJS Components have events?

### DISCUSSION THREAD

**Discussion by advancedsoftwarecanada** [GrapesJS/grapesjs#5710] [Ideas]
Loving Grapes, using it over at www.ScalarSites.com

Trying to build EASY to use components.... but I can't seem to get events listening for components. Any pointers? I want to click a button within the components to take an action.

![image](https://github.com/GrapesJS/grapesjs/assets/1661561/8ebae37a-6378-4352-9d17-e7c06d8143cb)

**Comment by mosh-tudor** [comment]
@advancedsoftwarecanada 
Check view/events: https://grapesjs.com/docs/modules/Components.html#view

**Reply by advancedsoftwarecanada** [reply]
Awesome this looks promising! Thank you.

**Reply by advancedsoftwarecanada** [reply]
OK So it looks like I need to use Traits, I have it working with DOM elements. Pretty cool... But it isnt clear if I have to "save" the data in a database, or the traits can be loaded simply from GrapesJS HTML?

**Reply by advancedsoftwarecanada** [reply]
OK. This can be closed. Traits were the answer. Its a bit of trickery to get it onto the DOM but it works. THX.
